adhocteam/Head-Start-TTADP
# [TTAHUB-110][TTAHUB-146]: Filter and Report menus a11y improvements

## Description of change

Various changes to filter and report menus to improve a11y specifically as it relates to keyboard navigation and screen readers.

Both Filter and ReportMenu components were given "menu" roles for the menus. The Filter component was given the "toolbar" role since it is a collection of controls. Event handlers for keyUp events were added to close menus and move focus back to the button that opened the menu.

When filters are added using the Add Filter button, focus is moved to the new filter. Labels for the filter as a whole were adjusted for better readability.

"Phrases" were adjusted on the DateRangePicker so it wouldn't use the words "checkin" or "checkout" in reference to picking start and end dates.

I created an `AriaLiveRegion` that is loaded in the DOM early on so we can have status-type messages read out to screen readers after interactions like applying filters. So when you apply a filter, a message like "1 filter applied" should be announced to the user.

Discussion around various expectations are on [the ticket, TTAHUB-110](https://ocio-jira.acf.hhs.gov/browse/TTAHUB-110). 

## How to test

Use a screenreader such as VoiceOver on macOS to navigate to and into Filter and Report menus using the keyboard. For VoiceOver, the [keyboard navigation](https://support.apple.com/guide/voiceover/navigation-commands-cpvokys04/10/mac/11.0) involves using the [VoiceOver modifier](https://support.apple.com/guide/voiceover/voiceover-modifier-unac048/10/mac/11.0), e.g. `Control-Option-` to navigate a page. Note that **tab navigation is not necessarily the same as screenreader navigation**.

Create, apply, and remove various filters. 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-110
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-146


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
